gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -54,206 +61,492 @@
+            ]
+            lof_combo_override = {"loftee_HC": ["HC"], "loftee_flags": ["with_flags"]}
+            filter_group_key_rename = {"lof_csq": "csq","loftee_HC": "loftee_labels"}
+
+        The function will generate the following filter combinations:
+
+        .. code-block:: python
+
+            [
+               # Combinations of all common filter keys and their possible values.
+                {},
+                {'capture': '1'},
+                {'capture': '2'},
+                {'variant_qc': 'pass'},
+                {'variant_qc': 'pass', 'capture': '1'},
+                {'variant_qc': 'pass', 'capture': '2'},
+
+                # Combinations of all requested common filter combinations with all
+                # possible other filter keys and values.
+                {'variant_qc': 'pass', 'max_af': '0.01'},
+                {'variant_qc': 'pass', 'csq': 'stop_gained'},
+                {'variant_qc': 'pass', 'capture': '1', 'max_af': '0.01'},
+                {'variant_qc': 'pass', 'capture': '1', 'csq': 'stop_gained'},
+
+                # Combinations of all requested common filter combinations with all
+                # requested LOF filter combination keys and their requested values.
+                {
+                    'variant_qc': 'pass', 'loftee_labels': 'HC',
+                    'loftee_flags': 'with_flags'
+                },
+                {
+                    'variant_qc': 'pass', 'csq': 'stop_gained', 'loftee_labels': 'HC',
+                    'loftee_flags': 'with_flags'
+                },
+                 {
+                    'variant_qc': 'pass', 'capture': '1', 'loftee_labels': 'HC',
+                    'loftee_flags': 'with_flags'
+                },
+                {
+                    'variant_qc': 'pass', 'capture': '1', 'csq': 'stop_gained',
+                    'loftee_labels': 'HC', 'loftee_flags': 'with_flags'
+                }
+            ]
+
+    :param all_sum_stat_filters: Dictionary of all possible filter types.
+    :param common_filter_combos: List of lists of common filter keys to use for creating
+        common filter combinations.
+    :param common_combo_override: Dictionary of filter groups and their options to
+        override the values in `all_sum_stat_filters` for use with values in
+        `common_filter_combos`.
+    :param lof_combos: List of loss-of-function keys in all_sum_stat_filters to use for
+        creating filter combinations.
+    :param lof_combo_override: Dictionary of filter groups and their options to override
+        the values in `all_sum_stat_filters` for use with values in `lof_combos`.
+    :param filter_group_key_rename: Dictionary to rename keys in `all_sum_stat_filters`,
+        `common_combo_override`, or `lof_combo_override` to final metadata keys.
+    :return: Dictionary of filter field to metadata.
+    """
+    all_sum_stat_filters = deepcopy(all_sum_stat_filters)
```

Comment:
```suggestion
    all_sum_stat_filters = deepcopy(all_sum_stat_filters)
    include_lof_filters = 'lof' in all_sum_stat_filters.get('csq_set', []) or 'lof_csq' in all_sum_stat_filters
```

Reply:
I think this is too restrictive, the user can decide what combinations they want